dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java

```diff
@@ -1810,6 +1808,10 @@ private ResourceInfo updateResourceInfo(ResourceInfo res) {
             resourceInfo = new ResourceInfo();
             // get resource from database, only one resource should be returned
             Resource resource = getResourceById(resourceId);
+            if (Objects.isNull(resource)) {
+                logger.error("resource not found, resourceId: {}", resourceId);
+                return null;
```

Comment:
can we directly throw the exception here?

Reply:
> can we directly throw the exception here?

I don't think this is feasible because if an exception is thrown here, there will be a failed queue that will retry the task indefinitely, causing the error log to refresh the screen.